### PR TITLE
Quiz questions fetching

### PR DESCRIPTION
**Pull request by vexonius**

### CODE REVIEW

File: QuizApp/Client/Quiz/Models/QuizQuestionsResponse.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+struct QuizQuestionsResponse: Decodable {
```
Comment (istaudacher) [inline QuizApp/Client/Quiz/Models/QuizQuestionsResponse.swift]: QuizSessionResponse

File: QuizApp/Client/Quiz/Models/QuizQuestionsResponse.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    let sessionId: String
+    let questions: [QuestionResponse]
+
+}
+
+struct QuestionResponse: Decodable {
```
Comment (istaudacher) [inline QuizApp/Client/Quiz/Models/QuizQuestionsResponse.swift]: QuizQuestionResponse

File: QuizApp/Presentation/Leaderboard/LeaderboardViewController.swift [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (istaudacher) [inline QuizApp/Presentation/Leaderboard/LeaderboardViewController.swift]: ?

File: QuizApp/Presentation/Coordinators/AppCoordinator.swift [hunk 4]
```diff
@@ -62,4 +62,8 @@ extension AppCoordinator: QuizCoordinatorProtocol {
         navigationController.pushViewController(quizDetailsViewController, animated: true)
     }
 
+    func routeBackFromLeaderBoard() {
```
Comment (istaudacher) [inline QuizApp/Presentation/Coordinators/AppCoordinator.swift]: Didn't you correct this?

File: QuizApp/Domain/Quiz/Models/EmptyRequest.swift [hunk 5]
```diff
@@ -0,0 +1,3 @@
+struct EmptyRequest: Encodable {
```
Comment (istaudacher) [inline QuizApp/Domain/Quiz/Models/EmptyRequest.swift]: Where is this used?

**Review by istaudacher — CHANGES_REQUESTED** [review]

File: QuizApp/Domain/Quiz/Models/EmptyRequest.swift (hunk 5 above)
Comment (vexonius) [reply QuizApp/Domain/Quiz/Models/EmptyRequest.swift]: Here, as I can not supply nil as encodable

```
func startQuiz(with id: Int) async throws -> QuizSessionResponse {
        try await baseNetworkClient.post(
            url: QuizEndpoints.startSession(id: id).path,
            params: [:],
            headers: [HeaderField.contentType.key: HeaderValue.defaultContentType.value],
            requestBody: EmptyRequest())
}
```

Comment (istaudacher) [reply QuizApp/Domain/Quiz/Models/EmptyRequest.swift]: Can't you change the function so that it accepts `Encodable?`?

**Review by istaudacher — APPROVED** [review]

**Pull request merged**